obspy/io/seiscomp/tests/test_event.py

```diff
@@ -304,15 +303,24 @@ def test_write_remove_events(self):
                           event_removal=True)
             filepath_cmp = \
                 os.path.join(self.path, 'qml-example-1.2-RC3_no_events.sc3ml')
-            self.assertTrue(filecmp.cmp(filepath_cmp, tf.name))
+            self._compare(tf, filepath_cmp)
 
     def test_read_and_write(self):
         filename = os.path.join(self.path, 'qml-example-1.2-RC3_write.sc3ml')
         catalog = read_events(filename)
 
         with NamedTemporaryFile() as tf:
             catalog.write(tf, format='SC3ML', validate=True)
-            self.assertTrue(filecmp.cmp(filename, tf.name))
+            self._compare(tf, filename)
+
+    def _compare(self, tf, filename):
+        tf.seek(0)
+        dat1 = tf.read().decode("utf-8").splitlines()
+        with open(filename, 'rb') as f:
+            dat2 = f.read().decode("utf-8").splitlines()
+        self.assertEqual(len(dat1), len(dat2))
+        for d1, d2 in zip(dat1, dat2):
+            self.assertEqual(d1, d2)
```

Comment:
This might be better placed under `core/utils/testing.py` since it seems to be generic to ASCII text file comparison and might be useful in other tests as well?